packages/wrangler/src/config/validation.ts

```diff
@@ -3103,6 +3114,44 @@ const validateConsumer: ValidatorFn = (diagnostics, field, value, _config) => {
 	return isValid;
 };
 
+const validatePipelineBinding: ValidatorFn = (diagnostics, field, value) => {
+	if (typeof value !== "object" || value === null) {
+		diagnostics.errors.push(
+			`"pipeline" bindings should be objects, but got ${JSON.stringify(
+				value
+			)}`
+		);
+		return false;
+	}
+	let isValid = true;
+	// Pipelin bindings must have a binding and a pipeline.
+	if (!isRequiredProperty(value, "binding", "string")) {
+		diagnostics.errors.push(
+			`"${field}" bindings should have a string "binding" field but got ${JSON.stringify(
```

Comment:
This pattern is used through this file.  I think the end result is that string interpolation of undefined it will show a message:
```
... binding should have a string "binding" field but got undefined
```
Which seems still like a reasonable outcome.